src/main/java/emissary/core/channels/LoggingInputStream.java

```diff
@@ -0,0 +1,135 @@
+package emissary.core.channels;
```

Comment:
Since this class isn't directly related to channels in any way, we should move it to the `emissary.util.io` package, which has other custom streams such `LoggingPrintStream` and `BufferedWriterOutputStream`
```suggestion
package emissary.util.io;
```